tiny_dnn/core/kernels/global_avepool_op_avx.h

```diff
@@ -0,0 +1,83 @@
+/*
+    Copyright (c) 2017, Taiga Nomi
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+
+namespace tiny_dnn {
+namespace kernels {
+
+#ifdef CNN_USE_AVX
```

Comment:
`global_avepool_op_avx` and `global_avepool_grad_op_avx` functions are referenced even when `CNN_USE_AVX` isn't defined. So if you reference them, you should keep them.

In `tiny_dnn/core/kernels/conv2d_op_avx.h` file, I divided interface function `conv2d_op_avx` and implementation function `avx_conv2d_5x5_kernel`.